fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java

```diff
@@ -81,9 +81,10 @@ public AggregatedMaterializedViewPushDownRewriter(MvRewriteContext mvRewriteCont
     public OptExpression doRewrite(MvRewriteContext mvContext) {
         OptExpression input = mvContext.getQueryExpression();
 
-        // try push down
-        OptExpression inputDuplicator = duplicateOptExpression(mvContext, input);
-        AggRewriteInfo rewriteInfo = process(inputDuplicator, AggregatePushDownContext.EMPTY);
+        OptExpression cloned = OptExpressionCloner.clone(input);
```

Comment:
OptExpressionCloner clone the OptExpression and its offsprings recursively and deeply, it seems time-consuming, why need perform this operartions?  for other CBO rules, only few OptExpressions themself  is mutated without cloning its expressions/predicates  and offspring OptExpressions deeply.

Reply:
yeah. you are right, it's time costing.

1. This is because `AggregatedMaterializedViewPushDownRewriter` rule may corrupt original OptExpression if agg after pushed down cannot be rewritten.
2. Cloning is just a stupid way to protect this to rollback to the original opt expression.

You are right, we can optimize this if we can ensure the agg push down and mv rewrite is `atomic` later.